### DISCUSSION TITLE
[Question] Typescript support?

### DISCUSSION THREAD

**Discussion by badstorm** [orchidjs/tom-select#131] [Q&A]
Hi, I'm trying to add TomSelect to my typescript project. Here my steps:
First with yarn I added tom-select: `yarn add tom-select`

Then in my .ts file i imported tom-select: `import TomSelect from 'tom-select'`

Now I try to initialized the plugin doing:
```
[].slice.call(document.querySelectorAll('.select-multiple')).map(function (item: HTMLInputElement) {
      new TomSelect(item, null);
  });
```

When i run my app i get 55 errors related to TomSelect. Here some of them:
- TS7016: Could not find a declaration file for module './contrib/microevent.js'. '...' implicitly has an 'any' type.
- TS7016: Could not find a declaration file for module '@orchidjs/sifter/dist/esm/sifter.js'. '...' implicitly has an 'any' type.
- TS2339: Property 'dataset' does not exist on type 'Element'
 
What's was wrong? I forgot some steps?
Thanks

**Comment by oyejorge** [comment]
Does your Typescript config have `strict = false`?

 [Related discussion](https://github.com/orchidjs/tom-select/discussions/100#discussioncomment-781684)

**Reply by badstorm** [reply]
Thanks for your reply, I just tryied added "strict: false" to my tsconfig.json (i'm working with webpack) and now i get only this error `Uncaught TypeError: tom_select_1.default is not a constructor` at line `new TomSelect(item, null);`

**Reply by badstorm** [reply]
I tryed to change `import TomSelect from 'tom-select'` with  `import TomSelect from 'tom-select/dist/esm/tom-select.complete.js'` and i get a compiler error `Could not find a declaration file for module 'tom-select/dist/esm/tom-select.complete.js'. '...' implicitly has an 'any' type` but i don't know why on the html page the control works.

**Reply by oyejorge** [reply]
Honestly have no idea why you'd get `Uncaught TypeError: tom_select_1.default ...` which Tom Select version are you using?

I wonder if it would make more sense to import the Typescript source directly with `import TomSelect from 'tom-select/src/tom-select.complete`

**Reply by badstorm** [reply]
Change `import TomSelect from 'tom-select/dist/esm/tom-select.complete.js'` with `import TomSelect from 'tom-select/src/tom-select.complete` get a lot of errors, i'll attach the [errors log](https://gist.github.com/badstorm/394681d906acdfc37dfff0b07d9e52c7) if can help:

**Comment by fnagel** [comment]
So, no TypeScript support? I'm getting `Uncaught TypeError: tom_select_1.default is not a constructor` errors...
